src/main/kotlin/org/rust/cargo/runconfig/command/RunClippyAction.kt

```diff
@@ -17,9 +16,8 @@ class RunClippyAction : RunCargoCommandActionBase(CargoIcons.CLIPPY) {
 
     override fun actionPerformed(e: AnActionEvent) {
         val project = e.project ?: return
-        val toolchain = project.toolchain ?: return
         val cargoProject = getAppropriateCargoProject(e.dataContext) ?: return
-        val channel = if (toolchain.isRustupAvailable) RustChannel.NIGHTLY else RustChannel.DEFAULT
+        val channel = cargoProject.rustcInfo?.version?.channel ?: RustChannel.DEFAULT
```

Comment:
Looks like we don't need set specific channel at all because it can lead to the following situation:
We have `1.31.0` and `stable` toolchains. `1.31.0` is default.
Also, we've installed clippy only for `1.31.0`.
And when we invoke this action, we'll launch `cargo +stable clippy` command and get `error: 'cargo-clippy' is not installed for the toolchain 'stable-...'` message

Reply:
Thank you for the suggestion, had pushed the changes suggested.

Could you explain what is the meaning of the value stored in `cargoProject.rustcInfo?.version` then?
I was under the impression that this is the version of the toolchain used in the current cargo project (and in this context selecting the clippy tool from a different toolchain sounded like a bad idea), but apparently I'm wrong.

Reply:
> Could you explain what is the meaning of the value stored in cargoProject.rustcInfo?.version then?

It's a version of rustc for current project stored after the latest project model refresh

> In this context selecting the clippy tool from a different (default) toolchain sounded like a bad idea, but apparently I'm wrong and this version is something else.

But your latest change is exactly about using default toolchain for the current project instead of nightly toolchain i.e. the same toolchain version as user uses for project compilation.

Reply:
Looks like `default` word is misleading here
In our case `default` channel means that we **don't** add any channel argument to the command line, i.e. we use **active** toolchain for this project

Reply:
Thanks for the explanations, now it's clearer.